src/internal/util/executeSchedule.ts

```diff
@@ -0,0 +1,44 @@
+import { Subscription } from '../Subscription';
+import { SchedulerAction, SchedulerLike } from '../types';
+
+export function executeSchedule(
+  parentSubscription: Subscription,
+  scheduler: SchedulerLike,
+  work: () => void,
+  delay: number,
+  repeat: true
+): void;
+export function executeSchedule(
+  parentSubscription: Subscription,
+  scheduler: SchedulerLike,
+  work: () => void,
+  delay?: number,
+  repeat?: false
+): Subscription;
+
+export function executeSchedule(
+  parentSubscription: Subscription,
+  scheduler: SchedulerLike,
+  work: () => void,
+  delay = 0,
+  repeat = false
+): Subscription | void {
+  const scheduleSubscription = scheduler.schedule(function (this: SchedulerAction<any>) {
+    work();
+    if (repeat) {
+      parentSubscription.add(this.schedule(null, delay));
```

Comment:
I can't see why it's necessary to add the action to the parent subscription if it repeats. It should already have been added and it won't be removed because it's repeating and the `unsubscribe` call - in the other branch of this `if` statement - is going to be skipped.

Reply:
Because someone can create their own Scheduler/Action implementations right now. (I hate exporting classes). And there's no guarantee they're properly reusing actions.

Reply:
Can check for that? Internal actions should be returning themselves as the subscription.

```ts
const repeatSubscription = this.schedule(null, delay);
if (repeatSubscription !== this) {
  parentSubscription.add(repeatSubscription);
}
```